api/src/test/java/org/openmrs/api/impl/DiagnosisServiceImplTest.java

```diff
@@ -330,5 +340,212 @@ public void purgeDiagnosis_shouldPurgeDiagnosis() {
+		diagnosisAttributeType.setCreator(diagnosis.getCreator());
+		diagnosisAttributeType.setDateCreated(diagnosis.getDateCreated());
+		diagnosisAttributeType.setRetired(false);
+		diagnosisAttributeType.setUuid("353af72e-bb6e-4ed9-a1bf-0d8106ac2c15");
+		diagnosisService.saveDiagnosisAttributeType(diagnosisAttributeType);
+		assertNotNull(diagnosisAttributeType.getDiagnosisAttributeTypeId(), "Newly Saved Diagnosis Attribute Type");
+		assertEquals(ORIGINAL_COUNT + 1, diagnosisService.getAllDiagnosisAttributeTypes().size());
+		DiagnosisAttributeType savedDiagnosisAttributeType = diagnosisService.getDiagnosisAttributeTypeByUuid("353af72e-bb6e-4ed9-a1bf-0d8106ac2c15");
+		assertEquals("Clinical Decision Support System", savedDiagnosisAttributeType.getName());
+		assertEquals(1, savedDiagnosisAttributeType.getMinOccurs());
+		assertEquals(5, savedDiagnosisAttributeType.getMaxOccurs());
+		assertEquals(diagnosis.getCreator(), savedDiagnosisAttributeType.getCreator());
+		assertEquals(diagnosis.getDateCreated(), savedDiagnosisAttributeType.getDateCreated());
+		assertEquals("353af72e-bb6e-4ed9-a1bf-0d8106ac2c15", savedDiagnosisAttributeType.getUuid());
+		assertEquals(false, savedDiagnosisAttributeType.getRetired());
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#saveDiagnosisAttributeType(DiagnosisAttributeType)
+	 */
+	@Test
+	public void saveDiagnosisAttributeType_shouldEditTheExistingDiagnosisAttributeType() {
+		DiagnosisAttributeType diagnosisAttributeType = diagnosisService.getDiagnosisAttributeTypeById(3);
+		assertEquals("Diagnostic Criteria", diagnosisAttributeType.getName());
+		diagnosisAttributeType.setName("Diagnosis Orientation");
+		diagnosisService.saveDiagnosisAttributeType(diagnosisAttributeType);
+		assertThat(diagnosisService.getDiagnosisAttributeTypeById(diagnosisAttributeType.getId()).getName(), is("Diagnosis Orientation"));
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnosisAttributeByUuid(String)
+	 */
+	@Test
+	public void getDiagnosisAttributeByUuid_shouldGetTheDiagnosisAttributeWithTheProvidedUuid() {
+		DiagnosisAttribute diagnosisAttribute = diagnosisService.getDiagnosisAttributeByUuid("31f7c3cd-699b-4ed3-af10-563e024cae76");
+		assertEquals("Testing Reference", diagnosisAttribute.getValueReference());
+		assertEquals(1, diagnosisAttribute.getId());
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#getDiagnosisAttributeByUuid(String)
+	 */
+	@Test
+	public void getDiagnosisAttributeByUuid_shouldReturnNullIfNoDiagnosisAttributeHasTheProvidedUuid() {
+		assertNull(diagnosisService.getDiagnosisAttributeByUuid("001612a5-9f2a-4a05-9384-755679f75647"));
+	}
+
+	/**
+	 * @see org.openmrs.api.DiagnosisService#save(Diagnosis)
+	 */
+	@Test
+	public void saveDiagnosis_shouldSaveTheDiagnosisWithTheProvidedAttributes() {
+		final String NAMESPACE = "namespace", FORMFIELD_PATH = "formFieldPath";
+		final DiagnosisAttributeType diagnosisAttributeType = Context.getDiagnosisService()
```

Comment:
Naming convention for constants: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-NamingConventions

Reply:
Did you see the above comment?